### PR TITLE
Add OpenAPI schema validation

### PR DESCRIPTION
**Pull request by bison**
This adds OpenAPI schema validation via `kubebuilder` comments. The code generation switches back to using the operator-sdk (version v0.5.0 or newer) to enable this.

### CODE REVIEW

**Comment by bison** [comment]
/hold cancel

Did some manual testing. Seems good to me.

File: install/0000_50_cluster-autoscaler-operator_01_clusterautoscaler.crd.yaml [hunk 1]
```diff
@@ -1,8 +1,7 @@
-# TODO(bison): Add schema.
----
 apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
+  creationTimestamp: null
```
Comment (frobware) [inline install/0000_50_cluster-autoscaler-operator_01_clusterautoscaler.crd.yaml]: Why do we need this?

Comment (ingvagabund) [reply install/0000_50_cluster-autoscaler-operator_01_clusterautoscaler.crd.yaml]: Generated automatically

Comment (bison) [reply install/0000_50_cluster-autoscaler-operator_01_clusterautoscaler.crd.yaml]: The CRDs are generated now, by operator-sdk. It adds that bit. We don't need it, but it also doesn't hurt anything.

**Pull request merged**